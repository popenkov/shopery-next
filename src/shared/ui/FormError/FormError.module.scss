.FormError {
  padding: 12px 20px;
  background-color: var(--status-danger);
  border-radius: 8px;
}

.FormErrorText {
  color: var(--front-primary-inversed);
}
