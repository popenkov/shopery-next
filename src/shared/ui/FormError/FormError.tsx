import { FC } from 'react';
import cn from 'classnames';
import { Text } from '../Text';
import cls from './FormError.module.scss';

interface Props {
  text: string;
  className?: string;
}

export const FormError: FC<Props> = ({ text, className }) => {
  return (
    <div className={cn(cls.FormError, className)}>
      <Text className={cls.FormErrorText}>{text}</Text>
    </div>
  );
};
