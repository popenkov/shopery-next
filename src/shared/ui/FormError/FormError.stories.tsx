import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';

import { AppRouteNames } from '@/shared/model';
import { FormError } from './FormError';

const meta = {
  title: 'shared/FormError',
  component: FormError,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: { text: 'error' },
  argTypes: {},
} satisfies Meta<typeof FormError>;

export default meta;
type TStory = StoryObj<typeof meta>;

export const FormErrorComponent: TStory = {
  decorators: [
    (Story) => (
      <div>
        <Story />
      </div>
    ),
  ],
  args: {
    text: 'Error text',
  },
};
