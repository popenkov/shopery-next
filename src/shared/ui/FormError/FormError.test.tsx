import React from 'react';
import { render } from '@testing-library/react';

import { FormError } from './FormError';

describe('FormError component', () => {
  it('renders correctly', () => {
    const text = 'Error message';
    const { container } = render(<FormError text={text} />);
    expect(container).toMatchSnapshot();
  });

  it('renders error text correctly', () => {
    const text = 'Error message';
    const { getByText } = render(<FormError text={text} />);
    const errorText = getByText(text);
    expect(errorText).toBeInTheDocument();
  });

  it('applies custom class correctly', () => {
    const text = 'Error message';
    const customClass = 'custom-error-class';
    const { container } = render(<FormError text={text} className={customClass} />);
    const errorElement = container.querySelector('.custom-error-class');
    expect(errorElement).toBeInTheDocument();
  });
});
