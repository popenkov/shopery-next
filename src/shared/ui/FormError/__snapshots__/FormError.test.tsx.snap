// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormError component renders correctly 1`] = `
<div>
  <div
    class="FormError"
  >
    <p
      class="text FormErrorText primary body_m left regular"
    >
      Error message
    </p>
  </div>
</div>
`;
